tests/micro/common/test_mlperftiny.py

```diff
@@ -0,0 +1,295 @@
+        model_tar_path = temp_dir / "model.tar"
+        export_model_library_format(module, model_tar_path)
+        workspace_size = mlf_extract_workspace_size_bytes(model_tar_path)
+
+        extra_tar_dir = tvm.contrib.utils.tempdir()
+        extra_tar_file = extra_tar_dir / "extra.tar"
+        with tarfile.open(extra_tar_file, "w:gz") as tf:
+            with tempfile.TemporaryDirectory() as tar_temp_dir:
+                model_files_path = os.path.join(tar_temp_dir, "include")
+                os.mkdir(model_files_path)
+                header_path = generate_c_interface_header(
+                    module.libmod_name,
+                    [model_info["input_name"]],
+                    [model_info["output_name"]],
+                    [],
+                    {},
+                    [],
+                    0,
+                    model_files_path,
+                    {},
+                    {},
+                )
+                tf.add(header_path, arcname=os.path.relpath(header_path, tar_temp_dir))
+
+            create_header_file(
+                "output_data",
+                np.zeros(
+                    shape=model_info["output_shape"],
+                    dtype=model_info["output_dtype"],
+                ),
+                "include/tvm",
+                tf,
+            )
+
+        input_total_size = 1
+        input_shape = model_info["input_shape"]
+        for i in range(len(input_shape)):
+            input_total_size *= input_shape[i]
+
+        template_project_path = pathlib.Path(tvm.micro.get_microtvm_template_projects(platform))
+        project_options.update(
+            {
+                "extra_files_tar": str(extra_tar_file),
+                "project_type": project_type,
+                "board": board,
+                "compile_definitions": [
+                    f"-DWORKSPACE_SIZE={workspace_size + 512}",  # Memory workspace size, 512 is a temporary offset
+                    # since the memory calculation is not accurate.
+                    f'-DTARGET_MODEL={MLPERF_TINY_MODELS[model_name]["index"]}',  # Sets the model index for project compilation.
+                    f"-DTH_MODEL_VERSION=EE_MODEL_VERSION_{model_name.upper()}01",  # Sets model version. This is required by MLPerfTiny API.
+                    f"-DMAX_DB_INPUT_SIZE={input_total_size}",  # Max size of the input data array.
+                ],
+            }
+        )
+
+        if model_name != "ad":
+            project_options["compile_definitions"].append(
+                f'-DOUT_QUANT_SCALE={model_info["quant_output_scale"]}'
+            )
+            project_options["compile_definitions"].append(
+                f'-DOUT_QUANT_ZERO={model_info["quant_output_zero_point"]}'
+            )
+
+        project = tvm.micro.project.generate_project_from_mlf(
+            template_project_path, workspace_dir / "project", model_tar_path, project_options
+        )
+        project.build()
```

Comment:
is any validation of either the prediction or the running time results done for either the host-driven or mlperftiny tests?

Reply:
nevermind, I just reread your summary, 😁, but maybe add docstring with that info...

Reply:
yeah, we are testing accuracy here. I have added that to the doc string